bigquery_magics/config.py

```diff
@@ -26,7 +27,151 @@ def _get_default_credentials_with_project():
     return pydata_google_auth.default(scopes=_SCOPES, use_local_webserver=False)
 
 
-@dataclass
+class MissingReason(enum.Enum):
+    """Provides a way to disambiguate why an option is missing.
+
+    This is used instead of a None value to allow for custom validation and
+    docs generation.
+    """
+
+    # These missing reasons are because the engine doesn't support the feature,
+    # for example, use_rest_api on the bigframes engine. In this case, raise
+    # if the user has set it, as the magics would otherwise act in a way
+    # contrary to that in which the user explicitly requested.
+    NOT_SUPPORTED_BY_ENGINE_INFEASIBLE = enum.auto()
+
+    # This is like the above, but is theoretically possible. Include a call to
+    # action to reach out to the bigframes team if this is a feature they
+    # would like to use.
+    NOT_SUPPORTED_BY_ENGINE_BUT_POSSIBLE = enum.auto()
+
+    # This missing reason is for options that are magics-only and apply to all
+    # engines. For example, the destination variable is a magics-only setting
+    # and doesn't affect how queries are handled by the engine.
+    ENGINE_UNIVERSAL = enum.auto()
```

Comment:
Per our discussion offline about the spreadsheet, let's add an additional option for `SUPPORTED_BY_ENGINE_BUT_NO_OPTION` for things where there's some support for it in the engine (e.g. by setting the `cell_arg`), but there's no global option to override it.

```suggestion
    ENGINE_UNIVERSAL = enum.auto()
    
    # Similar to ENGINE_UNIVERSAL, these options _do_ work, but there
    # isn't a way to globally override them. Use this instead of ENGINE_UNIVERSAL
    # for things that really do require some explicit support in the engine (e.g. query parameters)
    # but don't have a way to set them for all queries.
    SUPPORTED_BY_ENGINE_BUT_NO_OPTION = enum.auto()
```